### ISSUE TITLE
Start thinking about address and road name grids

### ISSUE THREAD

**Issue by jidanni** [Zaczero/openstreetmap-ng#21]
Hi. Openstreetmap Next Generation is actually Openstreetmap Same Generation, unless you consider adding [address grid](https://www.jidanni.org/geo/house_numbering/grids/define.html) capability...

In fact address grids were a part of Last Generation -- paper maps, but were neglected in the jump to electronic maps.

Anyway how to add such capability... is something even Google or ESRI haven't thought about yet, so it's time to start thinking...

**Comment by Zaczero** [comment]
If some cities/countries depend on the address grid system then we surely need to do something about it. First, we would need to come up with a tagging schema, then, add support in map editors. Last step would be to integrate it in map rendering engine and search. But what's the role of OpenStreetMap-NG? Would we need to change anything on the website?

**Comment by jidanni** [comment]
Actually they don't "depend" on the grids. It just would be a great
enhancement.

And I'm not sure if they should have traditional OSM tags or not, or
should be something people drag around on their iD editor screen.

Or maybe they should be defined more like WGS84: at a different "level"
than tags.

OK, I just checked your README.md, and indeed maybe you folks have your
hands full already remaking OSM.

**Comment by Zaczero** [comment]
A tagging schema sounds like the only realistic solution for OSM, introducing anything else just for this one feature won't gain adoption (in my opinion). I think users should be able to map grid cells as areas, with each cell having specific parameters (more messy, simple to implement in editors and in end applications), or as lines/dots and depend on some algorithm (less messy in editors, hard to implement in end applications). I personally think areas would work best, and having a dedicated editor app for this feature shouldn't be hard to make. It would also be a good opportunity to improve the iD/Rapid filtering system.

**Comment by jidanni** [comment]
I get it! Have the user "Please enter at least three points"
sort of like https://mapwarper.net/
but under the hood using exactly the software and example at
https://gdal.org/programs/gdaltransform.html#ground-control-points
where the user has clicked on the first street corner,
and entered 0, 0, then the second corner, and entered, 0, -500,
oops, I meant 0, 500 S (more user friendly)...

Now of course they need to click on what cities and towns or counties their new
grid should cover. Nearby ones would appear in the sidebar...

And after they are finished, poof, the entire grid appears on the map,
there in the editor at least.

I see you are from PL but amazingly still understand North American
grids.

OK, your tips have given me a giant boost. (And maybe you should
co-author my proposed SOTM paper with me. It is missing any such user
interface ideas.

**Comment by Zaczero** [comment]
I am from Poland but I do understand that different countries have different needs - this is the first time I'm learning about address grids :D

I can prepare some user interface concepts for you to use. I'll share them here within few days.

**Comment by jidanni** [comment]
Here's the storyline for you!
https://www.jidanni.org/geo/house_numbering/grids/us/tx/ochiltree/perryton/perryton.html

**Comment by RedAuburn** [comment]
maybe i'm mis-understanding, but isn't this just https://wiki.osm.org/wiki/Tag:junction=yes ?

**Comment by jidanni** [comment]
I enhanced it with line grids in the picture and fluffier story.

**Comment by Zaczero** [comment]
Worries not. You can also get some traction by making a good blog/diary post :-) I'll prepare the designs nonetheless.

**Comment by Zaczero** [comment]
![image](https://github.com/Zaczero/openstreetmap-ng/assets/10835147/d2c2b406-65e6-4be0-b39e-74676d26cb33)

1. Select the area by picking three points; this will generate an outline.
2. Input the horizontal and vertical sizes to generate a grid.
3. Input min/max values for each edge to automatically fill the cells with interpolated values.
4. Click submit to transform the cells into editable areas with autocomplete tagging.
5. You can then manually edit each cell, change their size, merge, or delete (just like in iD/JOSM).
6. Save changes to upload the address grid to OpenStreetMap.

**Comment by jidanni** [comment]
OK! May I suggest
1. You use Salt Lake City as the background to practice on. https://wiki.openstreetmap.org/wiki/Utah/Naming_Conventions/Activity . Its street names _are_ the grid numbers!
2. Never show values in the center of cells. Any values can only belong to the edges of cells. (Unless you make a dot at the center of the cell.)

**Comment by jidanni** [comment]
In https://www.jidanni.org/geo/house_numbering/grids/define.html#oak_park_il I made a template you can try. It's just a makefile where you fill in some data for some town and run "make ..." and presto, you are staring at the grid.

The idea is the user would be clicking on the screen, generation items that go into this form, which is then run when the user clicks  your submit button, etc.

**Comment by LaoshuBaby** [comment]
> 6\. Save changes to upload the address grid to OpenStreetMap.

This might be better off as a standalone editor rather than being provided directly on the OSM website?

**Comment by Zaczero** [comment]
Yes, this is currently not being planned as the direct OSM website feature. I'm just helping with an off-project issue :wink:. I will get back with the revised visualization.
